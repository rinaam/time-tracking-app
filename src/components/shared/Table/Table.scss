@import '../../../styles/colors.scss';
@import '../../../styles/spacing.scss';
@import '../../../styles/variables.scss';


.table-wrapper {
  width: 100%;
  border: $border-primary;
  margin: $spacing-xl 0;
  background-color: $background-white;

  thead {
    border-bottom: $border-primary;
  }

  th {
    padding: $spacing-xl;
    color: $label-text;
    border-right: $border-primary;
  }

  td {
    padding: $spacing-xl;
    text-align: center;
    border-right: $border-primary;
    border-bottom: $border-primary;
  }

  tbody {
    tr {
      &:hover {
        background-color: $background-hover;
      }
    }
  }
}