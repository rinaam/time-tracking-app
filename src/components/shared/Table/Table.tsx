import React from 'react';
import './Table.scss';

interface TableHeadField {
  text: string;
}

export interface TableBodyField {
  text?: string | number;
  component?: React.ReactNode;
  id: string;
}

interface TableProps {
  tableHead: TableHeadField[];
  tableBody: TableBodyField[][];
}

const Table: React.FC<TableProps> = ({ tableHead, tableBody }) => (
  <table className="table-wrapper">
    <thead>
      <tr>
        {tableHead.map((field) => (
          <th key={field.text}>{field.text}</th>
        ))}
      </tr>
    </thead>
    <tbody>
      {tableBody.map((row) => (
        <tr key={`row-${row[0].id}`} data-testid={`row-${row[0].id}`}>
          {row.map((field) => {
            if (field.text) {
              return (
                // eslint-disable-next-line react/no-danger
                <td key={field.id} dangerouslySetInnerHTML={{ __html: field.text as string }} />
              );
            }
            return <td key={field.id}>{field.component}</td>;
          })}
        </tr>
      ))}
    </tbody>
  </table>
);

export default Table;
