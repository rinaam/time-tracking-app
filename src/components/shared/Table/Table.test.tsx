import React from 'react';
import { render } from '@testing-library/react';
import Table from './Table';

it('should match snapshoot', () => {
  const tableHead = [{ text: 'title' }, { text: 'title1' }];
  const tableBody = [
    [
      { text: 'title', id: '1' },
      { text: 'title1', id: '2' },
    ],
  ];

  const wrapper = render(<Table tableHead={tableHead} tableBody={tableBody} />);
  expect(wrapper.asFragment()).toMatchSnapshot();
});
