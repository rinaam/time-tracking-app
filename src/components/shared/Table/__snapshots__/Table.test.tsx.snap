// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshoot 1`] = `
<DocumentFragment>
  <table
    class="table-wrapper"
  >
    <thead>
      <tr>
        <th>
          title
        </th>
        <th>
          title1
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        data-testid="row-1"
      >
        <td>
          title
        </td>
        <td>
          title1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
